sdk/servicebus/Microsoft.Azure.ServiceBus/src/ExceptionReceivedContext.cs

```diff
@@ -6,23 +6,30 @@
 namespace Microsoft.Azure.ServiceBus
 {
     using System;
+    using System.Collections.Generic;
 
     /// <summary>Context provided for <see cref="ExceptionReceivedEventArgs"/> exception raised by the client.</summary>
     public class ExceptionReceivedContext
     {
         /// <summary>Initializes a new instance of the <see cref="ExceptionReceivedContext" /> class.</summary>
+        /// <param name="messages">The messages associated with the exception, if applicable.</param>
         /// <param name="action">The action associated with the exception.</param>
         /// <param name="endpoint">The endpoint associated with the exception.</param>
         /// <param name="entityPath">The entity path associated with the exception.</param>
         /// <param name="clientId">The Client Id can be used to associate with the <see cref="QueueClient"/>, <see cref="SubscriptionClient"/>, <see cref="MessageSender"/> or <see cref="MessageReceiver"/> that encountered the exception.</param>
-        public ExceptionReceivedContext(string action, string endpoint, string entityPath, string clientId)
+        public ExceptionReceivedContext(Message[] messages, string action, string endpoint, string entityPath, string clientId)
         {
+            Messages = messages;
```

Comment:
The rest of the arguments are tested against null. Should messages be confirmed not to be null _and_ contain at least one message?

Reply:
No, there are situations where the raised exception cannot be attributed or associated with any messages. https://github.com/brandondahler/azure-sdk-for-net/blob/1c09dca9ce72bc08c94042cad308ab5cad830302/sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageReceivePump.cs#L117 is an a example of this situation.